public/js/index.js

```diff
@@ -1430,7 +1430,13 @@ $(document).ready(function () {
 
     // Highlight JS
     if (typeof hljs != 'undefined') {
-        hljs.initHighlightingOnLoad();
+        $('pre code').each(function (index, element) {
+            var worker = new Worker('/js/highlight-worker.js');
```

Comment:
Maybe we can only have one worker by putting it before the each.

Reply:
This will need to keep reference of element to update but will limit multiple worker loading

Reply:
@sapk - Sure, I've switched to that approach in https://github.com/go-gitea/gitea/pull/3591/commits/65605cda5986dca618374391aab734476097e27f - it now keeps hold of the array of code elements and passes the index into that array around so that we can update the right element when we get the highlighted html back. 

I also fixed a typo in the filename of the worker script.